### ISSUE TITLE
Changing the impress.js behaviour in relation to the backbutton?

### ISSUE THREAD

**Issue by torbjornlager** [impress/impress.js#408]
I have just started using impress.js (https://github.com/bartaz/impress.js/), and really like that it's so tiny and self-contained. ;-). I have a question about how to make use of it for my purposes though. To understand my use case, please assume the following:
1. P is an impress.js presentation
2. P contains two slides S and S´
3. If S is displayed, and if a presenter presses the right-arrow key, S´ becomes displayed.
4. S also contains a link L such that if the user clicks it, the browser will first unload impress.js and then display the page that L links to.

Now, my question is as follows: Would it be possible to make sure that if a presenter
   a) selects L (resulting in the unloading of impress.js), and then
   b) selects the browser's backbutton
then S´ rather than S will be displayed? As things now works S would first be displayed again, only ALLOWING a subsequent transition to S', subject to a decision by the presenter. This may often not be the behaviour we want.

If this would be possible (and that's what I'm asking) then wouldn't it be a great way to link impress.js presentations with the rest of the Web? It seems to me that it would enable the presenter of such an "extended presentation" to use the arrows-key and the backtbutton to traverse it in very nearly the same way as if it the whole thing was an ordinary impress.js presentation.

What am I missing?

Best regards,
Torbjörn Lager

**Comment by henrikingo** [comment]
Hmm... I just tried this with the impress.js demo.
 - I navigate to slide 5, which has the url hash "#/its"
 - I click the link to prezi.com
 - I press back button
 - I'm back at slide 5.

This works on Firefox and Chrome on Linux desktop. Is it different on some other browser?

**Comment by bartaz** [comment]
@henrikingo You are right, it works like that. But what was requested is different case.

Original issue is about having a linked page kind of as a part of the presentation, so when you go back to impress.js it shows **next** slide, not the same.
In case of impress.js demo it would be:

* I navigate to slide 5 with link to prezi.com
* I click on the link to prezi (to show how it looks like)
* I navigate back to impress.js and I expect it to show slide 6 to move forward with presentation.

As I mentioned above - it could be possible with some workarounds, but is quite tricky case to be covered in impress.js itself I believe.

**Comment by henrikingo** [comment]
Ah, thanks for clarification. Ok, I see now that's what the original description says!

In that case this is much more a special case. But just as an idea, as we're talking:

 - when the link is clicked, this is captured by javascript before navigating to new page.
 - javascript stores a variable in a cookie
 - when returning to presentation, impress.js will check that we returned to slide S, and will notice the variable set in the cookie, and therefore immediately calls next()

(Sounds like a good feature for a plugin, once we support plugins.)

**Comment by FagnerMartinsBrack** [comment]
Tl;DR of this use case is:

Make it possible for the presenter to click in an external link during a presentation, and when it returns back to the page (pressing the `back` button) it should go to the `next` step.

**Comment by henrikingo** [comment]
Hi @torbjornlager 

As I'm reviewing open issues, I came back to this. I'm thinking the following link does roughly what you expect (and in my opinion is a more standard way to do the same):

    <a href="link/to/other/page.html" target="_blank" onclick="impress().next()">visit external page</a>

The above will open your link in a new tab, while the impress.js presentation stays open in the old tab. Also, when you click on the link, the presentation will already flip to the next slide. When you're finished looking at the external page, you close that tab, and return to impress.js, which is ready with the next slide for you.

As this solution will work with impress.js as it is, and doesn't require any code changes, I will close this issue. But you're welcome to comment more if this didn't fully answer your need.
